### PR TITLE
Feature/disentangled vae

### PR DESCRIPTION
**Pull request by mini96**
- Added Beta VAE model 
- Amended evaluation script to accommodate for the model.
- Utilized code to generate and reconstruct loss metrics 
- Added a boiler plate for the disentanglement evaluation file

### CODE REVIEW

**Comment by pgoell** [comment]
- Fix conflicts to main
- Remove all checkpoints (git lfs has been disabled from github for my account...)

Do you need all experiment versions or is there one that is representative for all? If so remove except for one. Add markdown descriptions inside the vX dir for the experiment

File: vae_medmnist/evaluation/eval_disentaglement.py
Comment (pgoell) [inline vae_medmnist/evaluation/eval_disentaglement.py:1]: Did you even use this file?

*visualize_latent_space_traversal* shows a plot, which is not possible the way the code is run, works only in interactive mode. (i.e. notebooks) -> either fix or remove

*compute_metrics* is not implemented -> either implement or remove

---
**Remove the file from if you decide to remove both of these functions**

File: vae_medmnist/evaluation/evaluation.py [hunk 1]
```diff
@@ -116,6 +124,8 @@ def save_reconstructions(model, dataloader, device, save_path, descriptions):
             _, reconstructions, _, _ = model(inputs)
         elif isinstance(model, VAE):
             reconstructions, _, _ = model(inputs)
+        elif isinstance(model, BetaVAE):
```
Comment (pgoell) [inline vae_medmnist/evaluation/evaluation.py:127]: redundant condition, merge with condition on condition before

File: vae_medmnist/evaluation/evaluation.py [hunk 2]
```diff
@@ -172,11 +190,13 @@ def save_reconstructions(model, dataloader, device, save_path, descriptions):
...
     elif hparams['model'] == 'cvae':
         model = CVAE.load_from_checkpoint(f"{hparams['checkpoint_dir']}/cvae-best-checkpoint.ckpt")
+    elif hparams['model'] == 'beta_vae':
+        model = BetaVAE.load_from_checkpoint(f"{hparams['checkpoint_dir']}/betavae-best-checkpoint-v4.ckpt")
     else:
         model = VAE.load_from_checkpoint(f"{hparams['checkpoint_dir']}/best-checkpoint.ckpt")
     model.to(args.device)
 
-    if hparams['datasets'] == 'tissuemnist':
+    if hparams['dataset'] == 'tissuemnist':
```
Comment (pgoell) [inline vae_medmnist/evaluation/evaluation.py:199]: no, param is datasets

File: vae_medmnist/evaluation/evaluation.py [hunk 3]
```diff
@@ -172,11 +190,13 @@ def save_reconstructions(model, dataloader, device, save_path, descriptions):
         model = ResNetVAE.load_from_checkpoint(f"{hparams['checkpoint_dir']}/best-checkpoint.ckpt")
     elif hparams['model'] == 'cvae':
         model = CVAE.load_from_checkpoint(f"{hparams['checkpoint_dir']}/cvae-best-checkpoint.ckpt")
+    elif hparams['model'] == 'beta_vae':
+        model = BetaVAE.load_from_checkpoint(f"{hparams['checkpoint_dir']}/betavae-best-checkpoint-v4.ckpt")
```
Comment (pgoell) [inline vae_medmnist/evaluation/evaluation.py:194]: I'd rather you'd only use one model as before and dont add version number to file name. If you create a vX folder, you should only keep the best model anyway

File: vae_medmnist/evaluation/evaluation.py [hunk 4]
```diff
@@ -147,13 +164,14 @@ def save_reconstructions(model, dataloader, device, save_path, descriptions):
...
     import yaml
-    from medmnist.dataset import TissueMNIST
+    from medmnist.dataset import TissueMNIST, ChestMNIST
 
     from vae_medmnist.dataloader.accumulated_dataset import AccumulatedMedMNIST
     from vae_medmnist.dataloader.medmnist_datamodule import MedMNISTDataModule
     from vae_medmnist.models.cvae import CVAE
     from vae_medmnist.models.resnet_vae import ResNetVAE
     from vae_medmnist.models.vae import VAE
+    from vae_medmnist.models.beta_vae import BetaVAE
```
Comment (pgoell) [inline vae_medmnist/evaluation/evaluation.py:174]: nit: import oder, beta_vae should be imported before cvae due to naming

File: vae_medmnist/evaluation/evaluation.py [hunk 5]
```diff
@@ -186,6 +206,10 @@ def save_reconstructions(model, dataloader, device, save_path, descriptions):
 
     datamodule.setup()
 
+    # Try to get labels from an instance of the dataset class
+    dataset_instance = datasetclass(split='train', download=True)
```
Comment (pgoell) [inline vae_medmnist/evaluation/evaluation.py:210]: remove, if you want a single dataset you can call the accumulated dataset with a list of ['datasetname'] and query the labers via the class

File: vae_medmnist/evaluation/evaluation.py [hunk 6]
```diff
@@ -147,13 +164,14 @@ def save_reconstructions(model, dataloader, device, save_path, descriptions):
 
     import pandas as pd
     import yaml
-    from medmnist.dataset import TissueMNIST
+    from medmnist.dataset import TissueMNIST, ChestMNIST
```
Comment (pgoell) [inline vae_medmnist/evaluation/evaluation.py:167]: ChestMNIST unused import?

File: vae_medmnist/evaluation/evaluation.py [hunk 7]
```diff
@@ -1,6 +1,14 @@
 import matplotlib.pyplot as plt
 import torch
 
+def get_labels(dataset_class):
```
Comment (pgoell) [inline vae_medmnist/evaluation/evaluation.py:4]: See comment down below on how to get labels for single dataset

File: vae_medmnist/models/components.py [hunk 8]
```diff
@@ -1,3 +1,5 @@
+"""components.py"""
```
Comment (pgoell) [inline vae_medmnist/models/components.py:1]: no, if you add a file docstring, at least do a meaningful one

File: vae_medmnist/models/components.py [hunk 9]
```diff
@@ -44,6 +46,8 @@ def dec_conv3x3(in_planes, out_planes, scale: int = 1):
     return nn.Sequential(Interpolate(scale_factor=scale), conv3x3(in_planes, out_planes))
 
 
+
```
Comment (pgoell) [inline vae_medmnist/models/components.py:49]: remove unnecessary additional lines. only 2 lines separation

File: vae_medmnist/models/components.py [hunk 10]
```diff
@@ -111,7 +115,6 @@ def forward(self, x):
 
         return out
 
-
```
Comment (pgoell) [inline vae_medmnist/models/components.py:114]: 2 lines separation

File: vae_medmnist/models/components.py [hunk 11]
```diff
@@ -243,6 +246,8 @@ def forward(self, x):
 
         x = self.conv1(x)
         return x
+    
+
```
Comment (pgoell) [inline vae_medmnist/models/components.py:250]: 2 lines separation

File: vae_medmnist/models/components.py
Comment (pgoell) [inline vae_medmnist/models/components.py:1]: Since you didnt alter anything, revert your commits to this file

**Review by pgoell — CHANGES_REQUESTED** [review]
